when mixing Scala and Java, scalac doesn't recognize unqualified name of inherited enum

compilation fails if all source files are passed to scalac:
```scala
  ~/newbug> cat I.java 
interface I { public enum E { E1 } }
  ~/newbug> cat J.java 
class J implements I {
	public void foo(E e) { }
}
  ~/newbug> cat S.scala 
object S { (new J).foo(null) }
  ~/newbug> ~/scala/dists/latest/bin/scalac I.java J.java S.scala 
J.java:2: error: not found: type E
	public void foo(E e) { }
                        ^
one error found
```

but succeeds if javac is run first:
```scala
  ~/newbug> javac *.java
  ~/newbug> ~/scala/dists/latest/bin/scalac S.scala
```

the bug is very easy to work around by writing out the name more fully:
```scala
  ~/newbug> cat J.java 
class J implements I {
	public void foo(I.E e) { }
}
  ~/newbug> ~/scala/dists/latest/bin/scalac I.java J.java S.scala
```

I'm running r15746.

--- a/src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala
+++ b/src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala
@@ -492,13 +492,21 @@ trait ContextErrors extends splain.SplainErrors {
               }
             val semicolon = orEmpty(linePrecedes(qual, sel))(s"possible cause: maybe a semicolon is missing before `$nameString`?")
             val notAnyRef = orEmpty(ObjectClass.info.member(name).exists)(notAnyRefMessage(target))
-            val javaRules = orEmpty(owner.isJavaDefined && owner.isClass && !owner.hasPackageFlag) {
-              val (jtype, jmember) = cx.javaFindMember(target, name, _.isStaticMember)
-              orEmpty(jmember != NoSymbol) {
-                val more = sm"""Static Java members belong to companion objects in Scala;
-                  |they are not inherited, even by subclasses defined in Java."""
-                s"did you mean ${jtype.typeSymbol.fullName}.${jmember.name}? $more"
-              }
+            val javaRules = orEmpty(owner.isClass && !owner.hasPackageFlag) {
+              owner.baseClasses.iterator.filter(bc => bc.ne(ObjectClass) && bc.isJavaDefined)
+                .map { bc => cx.javaFindMember(bc.info, name, _.isStaticMember) match {
+                    case (_, NoSymbol) if name.isTermName =>
+                      cx.javaFindMember(bc.info, name.toTypeName, _.isStaticMember)
+                    case res => res
+                  }
+                }
+                .find(_._2 ne NoSymbol) match {
+                  case Some((jtype, jmember)) =>
+                    val more = sm"""Static Java members belong to companion objects in Scala;
+                      |they are not inherited, even by subclasses defined in Java."""
+                    s"did you mean ${jtype.typeSymbol.fullName}.${jmember.name}? $more"
+                  case _ => ""
+                }
             }
             List(companion, altStr, notAnyRef, semicolon, javaRules).filter("" != _).map("\n" + _).mkString
           }

--- a/test/files/neg/t1211.check
+++ b/test/files/neg/t1211.check
@@ -1,0 +1,11 @@
+test_2.scala:5: error: value E is not a member of J
+did you mean I.E? Static Java members belong to companion objects in Scala;
+they are not inherited, even by subclasses defined in Java.
+    j.f(j.E.E1)
+          ^
+test_2.scala:6: error: value E is not a member of object J
+did you mean I.E? Static Java members belong to companion objects in Scala;
+they are not inherited, even by subclasses defined in Java.
+    j.f(J.E.E1)
+          ^
+2 errors

--- a/test/files/neg/t1211/I.java
+++ b/test/files/neg/t1211/I.java
@@ -1,0 +1,4 @@
+
+interface I {
+	enum E { E1 }
+}

--- a/test/files/neg/t1211/J.java
+++ b/test/files/neg/t1211/J.java
@@ -1,0 +1,4 @@
+
+class J implements I {
+	void f(E e) { }
+}

--- a/test/files/neg/t1211/test_2.scala
+++ b/test/files/neg/t1211/test_2.scala
@@ -1,0 +1,8 @@
+
+object Test extends App {
+  val j = new J
+  println {
+    j.f(j.E.E1)
+    j.f(J.E.E1)
+  }
+}

--- a/test/files/neg/t1211b.check
+++ b/test/files/neg/t1211b.check
@@ -1,0 +1,11 @@
+test.scala:4: error: type K is not a member of object Test
+did you mean I.K? Static Java members belong to companion objects in Scala;
+they are not inherited, even by subclasses defined in Java.
+    new this.K
+             ^
+test.scala:5: error: type K is not a member of object Test
+did you mean I.K? Static Java members belong to companion objects in Scala;
+they are not inherited, even by subclasses defined in Java.
+    new Test.K
+             ^
+2 errors

--- a/test/files/neg/t1211b/I.java
+++ b/test/files/neg/t1211b/I.java
@@ -1,0 +1,4 @@
+
+class I {
+	static class K {}
+}

--- a/test/files/neg/t1211b/test.scala
+++ b/test/files/neg/t1211b/test.scala
@@ -1,0 +1,7 @@
+
+object Test extends I with App {
+  println {
+    new this.K
+    new Test.K
+  }
+}

--- a/test/files/neg/t1211c.check
+++ b/test/files/neg/t1211c.check
@@ -1,0 +1,11 @@
+test.scala:5: error: value E is not a member of J
+did you mean I.E? Static Java members belong to companion objects in Scala;
+they are not inherited, even by subclasses defined in Java.
+    j.f(j.E.E1)
+          ^
+test.scala:6: error: value E is not a member of object J
+did you mean I.E? Static Java members belong to companion objects in Scala;
+they are not inherited, even by subclasses defined in Java.
+    j.f(J.E.E1)
+          ^
+2 errors

--- a/test/files/neg/t1211c/I.java
+++ b/test/files/neg/t1211c/I.java
@@ -1,0 +1,4 @@
+
+interface I {
+	enum E { E1 }
+}

--- a/test/files/neg/t1211c/J.java
+++ b/test/files/neg/t1211c/J.java
@@ -1,0 +1,4 @@
+
+class J implements I {
+	void f(E e) { }
+}

--- a/test/files/neg/t1211c/test.scala
+++ b/test/files/neg/t1211c/test.scala
@@ -1,0 +1,8 @@
+
+object Test extends App {
+  val j = new J
+  println {
+    j.f(j.E.E1)
+    j.f(J.E.E1)
+  }
+}

